awscli/argprocess.py

```diff
@@ -369,118 +377,116 @@ class ParamShorthandDocGen(object):
-        s += self._generate_struct_list_scalar_docs(
-            cli_argument.argument_model.members)
-        return s
-
-    def _generate_struct_list_scalar_docs(self, members_dict):
-        scalar_params = list(self._get_scalar_params(members_dict))
-        list_params = list(self._get_list_params(members_dict))
-        pair = ''
-        for member_name, param in scalar_params:
-            pair += '%s=%s1,' % (member_name, param.type_name)
-        for member_name, param in list_params[:-1]:
-            param_type = param.member.type_name
-            pair += '%s=%s1,%s2,' % (member_name, param_type, param_type)
-        member_name, last_param = list_params[-1]
-        param_type = last_param.member.type_name
-        pair += '%s=%s1,%s2' % (member_name, param_type, param_type)
-        return pair
-
-    def _get_scalar_params(self, members_dict):
-        for key, value in members_dict.items():
-            if value.type_name in SCALAR_TYPES:
-                yield (key, value)
-
-    def _get_list_params(self, members_dict):
-        for key, value in members_dict.items():
-            if value.type_name == 'list':
-                yield (key, value)
-
-    def _has_enum_values(self, model):
-        return 'enum' in model.metadata
-
-    def _get_enum_values(self, model):
-        return model.metadata['enum']
-
-    def _docs_special_key_value_parse(self, cli_argument):
-        members = cli_argument.argument_model.members
-        if len(members) == 1 and 'Value' in members:
-            # Returning None will indicate that we don't have
-            # any examples to generate, and the entire examples section
-            # should be skipped for this arg.
+        docstring = self._handle_special_cases(cli_name, argument_model)
+        if docstring is self._DONT_DOC:
             return None
+        elif docstring:
+            return docstring
+
+        # Otherwise we fall back to the normal docgen for shorthand
+        # syntax.
+        stack = []
+        try:
+            if argument_model.type_name == 'list':
+                argument_model = argument_model.member
+                return self._shorthand_docs(argument_model, stack) + ' ...'
+            else:
+                return self._shorthand_docs(argument_model, stack)
+        except TooComplexError:
+            return ''
+
+    def _handle_special_cases(self, cli_name, model):
+        if model.type_name == 'list' and \
+                model.member.type_name == 'structure' and \
+                len(model.member.members) == 1:
+            member_name = list(model.member.members)[0]
+            return '%s %s1 %s2 %s3' % (cli_name, member_name,
+                                       member_name, member_name)
+        elif model.type_name == 'structure' and \
+                len(model.members) == 1 and \
+                'Value' in model.members and \
+                model.members['Value'].type_name == 'string':
+            return self._DONT_DOC
```

Comment:
Yes, this was actually a backwards compat thing so I didn't have to add any new tests for this.  This is covered by the pre-existing tests.